lib/Controller/RoomController.php

```diff
@@ -981,10 +1053,19 @@ public function demoteModerator($token, $participant) {
 			return new DataResponse([], Http::STATUS_FORBIDDEN);
 		}
 
-		if (!in_array($currentParticipant->getParticipantType(), [Participant::OWNER, Participant::MODERATOR], true)) {
+		if (!$currentParticipant->hasModeratorPermissions()) {
 			return new DataResponse([], Http::STATUS_FORBIDDEN);
 		}
 
+		if ($participant !== null) {
+			return $this->demoteUserToModerator($room, $participant);
+		}
+
+		return $this->demoteGuestToModerator($room, $sessionId);
+	}
+
+	protected function demoteUserToModerator(Room $room, string $participant): DataResponse {
```

Comment:
The name of this function should be changed to indicate that it is removing moderator status from the user not adding it.  demoteUserFromModerator would be a better name.